### PR TITLE
[Merged by Bors] - chore(Order): move some defs to a new file

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/ENat/Basic.lean [hunk 1]
```diff
@@ -24,22 +24,16 @@ and `Nat.cast` coercion. If you need to apply a lemma about `WithTop`, you may e
...
-/-- Extended natural numbers `ℕ∞ = WithTop ℕ`. -/
-def ENat : Type :=
-  WithTop ℕ
-deriving Zero,
-  -- AddCommMonoidWithOne,
-  CanonicallyOrderedCommSemiring, Nontrivial,
-  LinearOrder, Bot, Top, CanonicallyLinearOrderedAddCommMonoid, Sub,
+deriving instance Zero, CanonicallyOrderedCommSemiring, Nontrivial,
+  LinearOrder, Bot, CanonicallyLinearOrderedAddCommMonoid, Sub,
   LinearOrderedAddCommMonoidWithTop, WellFoundedRelation, Inhabited
```
Comment (grunweg) [inline Mathlib/Data/ENat/Basic.lean:29]: Do you want to remove `Inhabited`; you are also deriving that in `TypeTags` now.

File: Mathlib/Data/Countable/Basic.lean [hunk 2]
```diff
@@ -65,12 +65,19 @@ instance Sum.uncountable_inl [Uncountable α] : Uncountable (α ⊕ β) :=
 instance Sum.uncountable_inr [Uncountable β] : Uncountable (α ⊕ β) :=
   inr_injective.uncountable
 
-instance [Countable α] : Countable (Option α) :=
+instance Option.instCountable [Countable α] : Countable (Option α) :=
   Countable.of_equiv _ (Equiv.optionEquivSumPUnit.{_, 0} α).symm
 
+instance WithTop.instCountable [Countable α] : Countable (WithTop α) := Option.instCountable
```
Comment (grunweg) [inline Mathlib/Data/Countable/Basic.lean:71]: These are new. Do you mind splitting them into a dependent PR? I'll maintainer merge this one...

**Review by grunweg — COMMENTED** [review]
I saw this is draft, but took a look anyway: two minor comments from my side. (Is there something else you would like to add to this PR? If not, feel free to un-draft it.)

**Comment by urkud** [comment]
> I saw this is draft, but took a look anyway: two minor comments from my side. (Is there something else you would like to add to this PR? If not, feel free to un-draft it.)

@grunweg  Done.

**Comment by grunweg** [comment]
Thanks!
maintainer merge

**Pull request closed without merging**